src/main/java/org/rumbledb/compiler/TranslationVisitor.java

```diff
@@ -735,91 +665,81 @@ public Void visitPostFixExpr(JsoniqParser.PostFixExprContext ctx) {
                 throw new OurBadException("Unrecognized postfix expression found.");
             }
         }
-        this.currentExpression = mainExpression;
-        return null;
+        return mainExpression;
     }
 
     @Override
-    public Void visitPredicate(JsoniqParser.PredicateContext ctx) {
-        this.visitExpr(ctx.expr());
-        return null;
+    public Node visitPredicate(JsoniqParser.PredicateContext ctx) {
+        return this.visitExpr(ctx.expr());
     }
 
     @Override
-    public Void visitObjectLookup(JsoniqParser.ObjectLookupContext ctx) {
+    public Node visitObjectLookup(JsoniqParser.ObjectLookupContext ctx) {
         // TODO [EXPRVISITOR] support for ParenthesizedExpr | varRef | contextItemexpr in object lookup
         if (ctx.lt != null) {
-            this.currentExpression = new StringLiteralExpression(
+            return new StringLiteralExpression(
                     ValueTypeHandler.getStringValue(ctx.lt),
                     createMetadataFromContext(ctx)
             );
         } else if (ctx.nc != null) {
-            this.currentExpression = new StringLiteralExpression(ctx.nc.getText(), createMetadataFromContext(ctx));
+            return new StringLiteralExpression(ctx.nc.getText(), createMetadataFromContext(ctx));
         } else if (ctx.kw != null) {
-            this.currentExpression = new StringLiteralExpression(ctx.kw.getText(), createMetadataFromContext(ctx));
+            return new StringLiteralExpression(ctx.kw.getText(), createMetadataFromContext(ctx));
         } else if (ctx.pe != null) {
-            this.visitParenthesizedExpr(ctx.pe);
+            return this.visitParenthesizedExpr(ctx.pe);
         } else if (ctx.vr != null) {
-            this.visitVarRef(ctx.vr);
+            return this.visitVarRef(ctx.vr);
         } else if (ctx.ci != null) {
-            this.visitContextItemExpr(ctx.ci);
+            return this.visitContextItemExpr(ctx.ci);
         } else if (ctx.tkw != null) {
```

Comment:
Shall we refactor these 'else if's to 'if's?

Reply:
Nice, done.